### PR TITLE
Bug 1164421 - autocorrect "im" to "I'm" not "in"

### PR DESCRIPTION
**Pull request by davidflanagan**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=649026cb72aa899a99e8eef5d2336e104f37708e

File: apps/keyboard/js/imes/latin/latin.js [hunk 1]
```diff
@@ -762,6 +762,52 @@
...
+    // list. A word is an approximate match if all the letters match
+    // when we ignore case, apostrophes and hyphens. The assumption here
+    // is that the user is not expected to have to use the shift key or
+    // the alternate forms menu or the punctuation menus. If they type all
+    // the right letters, a matching word should have higher priority than
+    // a non-matching word. In particular this code ensures that "im" gets
+    // autocorrected to "I'm" instead of "in" and "id" gets autocorrected to
+    // "I'd" instead of "is". (Bug 1164421)
+    //
+    // XXX In Engish, apostrohes and hypens are the only punctuation that
```
Comment (RudyLu) [inline apps/keyboard/js/imes/latin/latin.js]: typo: `English`,  `apostrophes`, `hyphens`.

File: apps/keyboard/js/imes/latin/latin.js [hunk 2]
```diff
@@ -762,6 +762,52 @@
...
+    // simple.
+    //
+    // We loop backward through the suggestions, bumping any approximate
+    // match to the start of the list. This ensures that the highest ranked
+    // approximate match comes before lower ranked approximate matches,
+    // and all approximate matches come before non-matching suggestions.
+    //
+    var i = suggestions.length - 1;
+    var approximateMatches = 0;  // how many approximate matches we've bumped
+    var lcinput = input.toLowerCase();
```
Comment (RudyLu) [inline apps/keyboard/js/imes/latin/latin.js]: `lcInput`, or `lowerCaseInput`

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=79e987f19487990de3a53401be85866e464e6f49

**Comment by autolander** [comment]
http://docs.taskcluster.net/tools/task-graph-inspector/#jQsEm7pJRUW7pmgaD6TLOg

The pull request failed to pass integration tests. It could not be landed, please try again.

**Comment by autolander** [comment]
http://docs.taskcluster.net/tools/task-graph-inspector/#z7atNthxSXqxURMk2sIyHA

The pull request failed to pass integration tests. It could not be landed, please try again.

**Pull request merged**